Packs/AzureNetworkSecurityGroups/Integrations/AzureNetworkSecurityGroups/AzureNetworkSecurityGroups.py

```diff
@@ -0,0 +1,352 @@
+        security_rule_name: the name of the rule
+
+    Returns:
+        CommandResults for the rule
+    """
+    # We want to flatten the rules `properties` key as this is the more important key and we'd like
+    # to be able to display it nicely
+    if isinstance(rule_json, dict):
+        rule_json.update(rule_json.pop('properties', {}))
+    if isinstance(rule_json, list):
+        for rule in rule_json:
+            rule.update(rule.pop('properties', {}))
+
+    hr = tableToMarkdown(f"Rules {security_rule_name}", rule_json, removeNull=True)
+
+    return CommandResults(outputs_prefix='AzureNSG.Rules',
+                          outputs_key_field='id',
+                          outputs=rule_json,
+                          readable_output=hr)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+@logger
+def list_groups_command(client: AzureNSGClient) -> CommandResults:
+    """
+
+    Args:
+        client: The MSClient
+        args: args dictionary. Should be empty for this command.
+
+    Returns:
+        A detailed list of all network security groups
+    """
+    network_groups = client.list_network_security_groups()
+    network_groups = network_groups.get('value', [])
+
+    # The property value holds all the rules under the security group. This will be returned in a different command.
+    for group in network_groups:
+        group.pop('properties', '')
+
+    hr = tableToMarkdown('Network Security Groups', network_groups)
+    return CommandResults(outputs_prefix='AzureNSG.SecurityGroup',
+                          outputs_key_field='id',
+                          outputs=network_groups,
+                          readable_output=hr)
+
+
+@logger
+def list_rules_command(client: AzureNSGClient, security_group_name: str) -> CommandResults:
+    """
+
+    Args:
+        client: The MSclient
+        security_group_name: a comma-seperated list of security group names
+
+    Returns:
+        a list of  rules for the security group
+    """
+    security_groups = argToList(security_group_name)
+    rules: List = list()
+    for group in security_groups:
+        rules_returned = client.list_rules(group)
+        rules.extend(rules_returned.get('value', []))
+    return format_rule(rules, f"in {security_group_name}")
+
+
+@logger
+def delete_rule_command(client: AzureNSGClient, security_group_name: str, security_rule_name: str) -> str:
+    """
+
+    Args:
+        client: The MSClient
+        security_group_name: the name of the security group
+        security_rule_name: The name of the rule to delete
+
+    """
+
+    rule_deleted = client.delete_rule(security_group_name, security_rule_name)
+    if rule_deleted.status_code == 204:
+        raise ValueError(f"Rule {security_rule_name} not found.")
+    if rule_deleted.status_code == 202:
+        return f"Rule {security_rule_name} deleted."
+    else:
+        return f"Rule {security_rule_name} was not deleted. Got back the following reulst:\n{rule_deleted.content}"
```

Comment:
this should be an error (raise it as an exception)
```suggestion
        return f"Rule {security_rule_name} was not deleted. Got back the following result:\n{rule_deleted.content}"
```